File: app/src/main/java/uk/co/mishurov/fluid/ParticlesLib.java
package uk.co.mishurov.fluid;

import android.content.res.AssetManager;


public class ParticlesLib {
    static {
        System.loadLibrary( "particles" );
    }

    public static native void init();
    public static native void surface(int width, int height);
    public static native void touch(boolean is_down, float x, float y);
    public static native void step(float elapsedTime);
    public static native void rotate(int angle);
    public static native void settings(
        String fg_color, String bg_color, int iterations, int cursor_size
    );
    public static native void createAssetManager(AssetManager assetManager);
}

File: app/src/main/java/uk/co/mishurov/fluid/ParticlesLib.kt
package uk.co.mishurov.fluid

import android.content.res.AssetManager


object ParticlesLib {
    init {
        System.loadLibrary("particles")
    }

    external fun init()
    external fun surface(width: Int, height: Int)
    external fun touch(is_down: Boolean, x: Float, y: Float)
    external fun step(elapsedTime: Float)
    external fun rotate(angle: Int)
    external fun settings(
            fg_color: String, bg_color: String, iterations: Int, cursor_size: Int
    )

    external fun createAssetManager(assetManager: AssetManager)
}
